sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -833,10 +851,11 @@ template <typename T, int NElts = 1,
           lsc_data_size DS = lsc_data_size::default_size,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
           int N, typename AccessorTy, typename Toffset>
-__ESIMD_API std::enable_if_t<!std::is_pointer_v<AccessorTy> &&
-                                 std::is_integral_v<Toffset> &&
-                                 !std::is_same_v<Toffset, uint64_t>,
-                             __ESIMD_NS::simd<T, N * NElts>>
+__ESIMD_API std::enable_if_t<
+    !std::is_pointer_v<AccessorTy> &&
```

Comment:
Just a note here:
!std::is_pointer_v<AccessorTy> is a bad check (not because of your PR). It allows using write-only accessors lsc_gather() operations. Same/vice-versa is for read-only accessors usable in write operations in other places in memory.hpp here.

I created internal tracker for this issue.

Reply:
The much better check is used for copy_to() and copy_from(), e.g. copy_from():
```c++
detail::is_sycl_accessor_with<
                     AccessorT, accessor_mode_cap::can_read,
                     sycl::access::target::device>::value)
```